onnxruntime/core/framework/execution_frame.cc

```diff
@@ -190,6 +190,8 @@ ExecutionFrame::ExecutionFrame(const std::vector<int>& feed_mlvalue_idxs, const
   if (session_state.GetEnableMemoryPattern() && session_state.GetExecutionPlan()) {
     std::vector<TensorShape> input_shapes;
     bool all_tensors = true;
+    // Reserve mem to avoid re-allocation.
+    input_shapes.reserve(feeds.size());
```

Comment:
std::vector<TensorShape> input_shapes(feeds.size());

Reply:
This will cause push_back called below to increase the size of the vector beyond feeds.size().

Reply:
Then just use []. It's faster than anything else

Reply:
> This will cause push_back called below to increase the size of the vector beyond feeds.size().

You're right. Thanks

Reply:
This is different. Constructor will initialize initial values and put elements into it. Reserve is to reserve memory, but not actually insert values.